src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -3,25 +3,185 @@
 using ChainRulesCore
 using ITensors
 
+# Needed for defining the rule for `adjoint(A::ITensor)`
+# which currently doesn't work by overloading `ChainRulesCore.rrule`
+using ZygoteRules: @adjoint
+
 function ChainRulesCore.rrule(::typeof(getindex), x::ITensor, I...)
   y = getindex(x, I...)
   function getindex_pullback(ȳ)
-    x̄ = ITensor(inds(x))
+    # TODO: add definition `ITensor(::Tuple{}) = ITensor()`
+    # to ITensors.jl so no splatting is needed here.
+    x̄ = ITensor(inds(x)...)
     x̄[I...] = ȳ
     Ī = broadcast(_ -> NoTangent(), I)
     return (NoTangent(), x̄, Ī...)
   end
   return y, getindex_pullback
 end
 
-function ChainRulesCore.rrule(::typeof(*), x1::ITensor, x2::ITensor)
+# Specialized version in order to avoid call to `setindex!`
+# within the pullback, should be better for taking higher order
+# derivatives in Zygote.
+function ChainRulesCore.rrule(::typeof(getindex), x::ITensor)
```

Comment:
Is it only used on Itensor of a scalar?

Reply:
Yes. We use `A[]` and `scalar(A)` as two equivalent interfaces for extracting the number from an order-0 ITensor. `scalar(A)` calls `A[]` so this definition covers both.